indexer/scales.cpp

```diff
@@ -69,4 +72,17 @@ namespace scales
     // assume that feature is always visible in upper scale
     return (level == GetUpperScale() || std::max(r.SizeX(), r.SizeY()) > GetEpsilonForLevel(level));
   }
+
+  bool IsGoodOutlineForLevel(int level, Points const & poly)
+  {
+    // Areas and closed lines have the same first and last points.
+    // Hence the minimum number of outline points is 4.
+    if (poly.size() < 4)
+      return false;
```

Comment:
This logic is valid for areas, but I doubt for lines. So we skip all long A-B-A lines? Better keep A-B or every use-case is unique?

Maybe it is valid for isolines, but let us assume that we have say fence from OSM like A-B-A (mistakenly drawn).

Reply:
For non-isolines we can convert A-B-A to A-B somewhere around feature_maker (when source OSM ways are processed).
So at this point (after we start using IsGoodOutlineForLevel() in feature_sorter for non-isolines) there should be no such cases.